tests/test_metadata.py

```diff
@@ -655,3 +655,56 @@ def test_parse_until_resolved_skip_avoids_undefined_jinja(
         pytest.fail(
             "Undefined variable caused error, even though this build is skipped"
         )
+
+
+@pytest.mark.parametrize("have_variant", [True, False])
+def test_parse_until_resolved_missing_jinja_in_spec(
+    testing_metadata: MetaData,
+    tmp_path: Path,
+    have_variant: bool,
+) -> None:
+    (recipe := tmp_path / (name := "meta.yaml")).write_text(
+        """
+package:
+    name: dummy
+    version: 1.0.0
+
+build:
+    noarch: python
+    number: 0
+
+requirements:
+    host:
+        - python ={{ python_min }}
+    run:
+        - python >={{ python_min }}
+"""
+    )
+    (tmp_path / "conda_build_config.yaml").write_text(
+        """
+python_min:
+    - 3.6
+"""
+    )
+    testing_metadata._meta_path = recipe
+    testing_metadata._meta_name = name
+    if have_variant:
+        testing_metadata.config.variant = {"python_min": "3.6"}
+    else:
+        delattr(testing_metadata.config, "variant")
+        delattr(testing_metadata.config, "variant_config_files")
+        delattr(testing_metadata.config, "variants")
+
+    try:
+        testing_metadata.parse_until_resolved()
+        if not have_variant:
+            pytest.fail("Undefined variable did NOT cause spec parsing error!")
+        else:
+            print("parsed OK!")
+    except (Exception, SystemExit):
```

Comment:
Do we have a more concrete exception set?

Reply:
Not reliably no. I have already fixed a bug where when the exception handling in one part of the code changed, the test suite broke because a new kind of exception was raised in another part of the code. For now, any exception will work.